src/Microsoft.SqlTools.ServiceLayer/LanguageServices/BindingQueue.cs

```diff
@@ -286,86 +296,119 @@ private void ProcessQueue()
                             continue;
                         }
 
-                        bool lockTaken = false;
-                        try
-                        {                                                    
-                            // prefer the queue item binding item, otherwise use the context default timeout
-                            int bindTimeout = queueItem.BindingTimeout ?? bindingContext.BindingTimeout;
+                        var bindingContextTask = this.BindingContextTasks[bindingContext];
 
-                            // handle the case a previous binding operation is still running                                                 
-                            if (!bindingContext.BindingLock.WaitOne(queueItem.WaitForLockTimeout ?? 0))
-                            {
-                                queueItem.Result = queueItem.TimeoutOperation != null
-                                    ? queueItem.TimeoutOperation(bindingContext)
-                                    : null;
+                        // Run in the binding context task in case this task has to wait for a previous binding operation
+                        this.BindingContextTasks[bindingContext] = bindingContextTask.ContinueWith((task) =>
```

Comment:
Do we want this new task to run irrespective of what happened to original task? I suggest deliberately specify TaskContinuationOptions parameter even if default value as appropriate for us might be a good idea.